extensions/data-transfer/portability-data-transfer-mastodon/src/main/java/org/datatransferproject/transfer/mastodon/social/MastodonActivityExport.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright 2019 The Data Transfer Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.datatransferproject.transfer.mastodon.social;
+
+
+import static com.google.common.base.Preconditions.checkState;
+
+import com.google.common.base.Strings;
+import com.ibm.common.activitystreams.Activity;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import org.datatransferproject.spi.transfer.provider.ExportResult;
+import org.datatransferproject.spi.transfer.provider.ExportResult.ResultType;
+import org.datatransferproject.spi.transfer.provider.Exporter;
+import org.datatransferproject.spi.transfer.types.ContinuationData;
+import org.datatransferproject.transfer.mastodon.model.Account;
+import org.datatransferproject.transfer.mastodon.model.Status;
+import org.datatransferproject.types.common.ExportInformation;
+import org.datatransferproject.types.common.StringPaginationToken;
+import org.datatransferproject.types.common.models.social.SocialActivityContainerResource;
+import org.datatransferproject.types.transfer.auth.CookiesAndUrlAuthData;
+import com.ibm.common.activitystreams.Makers;
+
+public class MastodonActivityExport implements
```

Comment:
Add quick class java doc with what remains to be done or expanded on, format, etc.